### PR TITLE
Replace uses of SPV_AMD_shader_trinary_minmax extension

### PR DESCRIPTION
**Pull request by s-perron**
Part of #2814

### CODE REVIEW

File: source/opt/amd_ext_to_khr.cpp [hunk 1]
```diff
@@ -24,18 +24,114 @@ namespace opt {
...
+    return true;
+  };
+}
+
+// Returns a folding rule that replaces |mid(a,b,c)| by |clamp(a, min(b,c),
+// max(b,c)|. The three parameters are the opcode that correspond to the min,
+// max, and clamp operations for the type of the instruction being replaced.
+FoldingRule ReplaceTrinaryMid(GLSLstd450 min_opcode, GLSLstd450 max_opcode,
+                              GLSLstd450 clamp_opcode) {
+  return [min_opcode, max_opcode, clamp_opcode](
```
Comment (dnovillo) [inline source/opt/amd_ext_to_khr.cpp]: Likewise.  Maybe I'm missing something?

File: source/opt/amd_ext_to_khr.cpp [hunk 2]
```diff
@@ -24,18 +24,114 @@ namespace opt {
...
   analysis::Integer int_type(32, false);
   return ctx->get_type_mgr()->GetRegisteredType(&int_type);
 }
 
+// Returns a folding rule that replaces |op(a,b,c)| by |op(op(a,b),c)|, where
+// |op| is either min or max. |opcode| is the binary opcode in the GLSLstd450
+// extended instruction set that corresponds to the trinary instruction being
+// replaced.
+FoldingRule ReplaceTrinaryMinMax(GLSLstd450 opcode) {
+  return [opcode](IRContext* ctx, Instruction* inst,
```
[the remark this answers is not in the document]
Comment (s-perron) [reply source/opt/amd_ext_to_khr.cpp]: Yes it does.  We create a map from an opcode to a function object.  `ReplaceTrinaryMinMax` returns a lamba that that has been specialized for a particualr opcode.  That lamba will be inserted into the table.

File: source/opt/amd_ext_to_khr.cpp (hunk 1 above)
Comment (dnovillo) [reply source/opt/amd_ext_to_khr.cpp]: Ah, yes!  We are returning the lambda, not executing it.  Thanks.

File: source/opt/amd_ext_to_khr.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (s-perron) [reply source/opt/amd_ext_to_khr.cpp]: I believe I can turn it in a template function.

[the remark this answers is not in the document]
Comment (dnovillo) [reply source/opt/amd_ext_to_khr.cpp]: No worries.

[the remark this answers is not in the document]
Comment (s-perron) [reply source/opt/amd_ext_to_khr.cpp]: Fixed along with other function in the same file.

**Pull request merged**